### PR TITLE
fix(ME-1668): enable tls in sqlproxy to support connector auth

### PR DESCRIPTION
**Pull request by btoonk**
# Description

Enable TLS in sqlproxy to support connector auth.

Fixes # ME-1668

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Tested on staging.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: internal/sqlauthproxy/postgres.go [hunk 1]
```diff
@@ -175,3 +201,33 @@ func (h postgresHandler) handleClientAuthRequest(serverSession *pgproto3.Backend
...
+		return tls.Certificate{}, err
+	}
+
+	cert, err := x509.CreateCertificate(rand.Reader, template, template,
+		priv.Public(), priv)
+	if err != nil {
+		return tls.Certificate{}, err
+	}
+
+	var outCert tls.Certificate
```
Comment (adrianosela) [inline internal/sqlauthproxy/postgres.go:228]: Clearer if built in-line e.g.

outcert := tls.Certificate{
    Certificate: []...,
    PrivateKey: ...,
}

File: internal/sqlauthproxy/postgres.go [hunk 2]
```diff
@@ -175,3 +201,33 @@ func (h postgresHandler) handleClientAuthRequest(serverSession *pgproto3.Backend
...
+			CommonName:   "sqlauthproxy",
+			Organization: []string{"border.com"},
+		},
+		NotBefore: now,
+		NotAfter:  now.AddDate(10, 0, 0),
+	}
+
+	priv, err := rsa.GenerateKey(rand.Reader, 2048)
+	if err != nil {
+		return tls.Certificate{}, err
```
Comment (adrianosela) [inline internal/sqlauthproxy/postgres.go:219]: Wrapped errors to distinguish from one another

File: internal/sqlauthproxy/postgres.go [hunk 3]
```diff
@@ -175,3 +201,33 @@ func (h postgresHandler) handleClientAuthRequest(serverSession *pgproto3.Backend
...
+		SerialNumber: big.NewInt(now.Unix()),
+		Subject: pkix.Name{
+			CommonName:   "sqlauthproxy",
+			Organization: []string{"border.com"},
+		},
+		NotBefore: now,
+		NotAfter:  now.AddDate(10, 0, 0),
+	}
+
+	priv, err := rsa.GenerateKey(rand.Reader, 2048)
```
Comment (adrianosela) [inline internal/sqlauthproxy/postgres.go:217]: Take key size as arg to function?

File: internal/sqlauthproxy/postgres.go [hunk 4]
```diff
@@ -175,3 +201,33 @@ func (h postgresHandler) handleClientAuthRequest(serverSession *pgproto3.Backend
 
 	return nil
 }
+
+func generateX509KeyPair() (tls.Certificate, error) {
+	now := time.Now()
+	template := &x509.Certificate{
+		SerialNumber: big.NewInt(now.Unix()),
```
Comment (adrianosela) [inline internal/sqlauthproxy/postgres.go:208]: UnixNano for more uniqueness? Or uuid?

File: internal/sqlauthproxy/postgres.go [hunk 5]
```diff
@@ -73,21 +89,23 @@ func newPostgresHandler(c Config) (*postgresHandler, error) {
...
+	startupMessage, c, err := h.handleClientStartup(c)
 	if err != nil {
 		log.Printf("sqlauthproxy: failed to handle client startup: %s", err)
 		return
 	}
 
+	clientConn := pgproto3.NewBackend(pgproto3.NewChunkReader(c), c)
+
 	if startupMessage == nil {
+		log.Printf("sqlauthproxy: failed to handle client startup")
```
Comment (adrianosela) [inline internal/sqlauthproxy/postgres.go:108]: I think we should stop logging with sqlauthproxy -- the customer does not know what that is.

**Review by adrianosela — APPROVED** [review]
Approved with comments.

File: internal/sqlauthproxy/postgres.go (hunk 1 above)
Comment (btoonk) [reply internal/sqlauthproxy/postgres.go]: done

File: internal/sqlauthproxy/postgres.go (hunk 2 above)
Comment (btoonk) [reply internal/sqlauthproxy/postgres.go]: done

File: internal/sqlauthproxy/postgres.go (hunk 3 above)
Comment (btoonk) [reply internal/sqlauthproxy/postgres.go:217]: Not important here. I only need a tls sessions. It's already in a secure tunnel.

File: internal/sqlauthproxy/postgres.go (hunk 4 above)
Comment (btoonk) [reply internal/sqlauthproxy/postgres.go:208]: SerialNumber does not matter.

**Review by adrianosela — APPROVED** [review]

**Pull request merged**